### PR TITLE
backend: add POST and GET for rooms endpoint

### PR DESCRIPTION
**Pull request by tierrif**
As for POST /api/rooms/:id and OPTIONS /api/rooms/:id, those will be done in another time.
This PR also includes a small test in the front-end for GET /api/rooms that prints the content to the console.

### CODE REVIEW

File: backend/main.go [hunk 1]
```diff
@@ -51,12 +51,27 @@ func main() {
...
+		}
+	})
+	// A / in the end means there's URL params
+	http.HandleFunc("/api/room/", func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == "GET" {
+			// GET /api/room/:id and GET /api/room/:id/leave
+			GetRoom(w, r)
+		} else if r.Method == "POST" {
+			// POST /api/room/:id
 			http.Error(w, errorJson("Not Implemented!"), http.StatusNotImplemented) // TODO
```
Comment (retrixe) [inline backend/main.go:72]: This should be 404, I don't think we are putting an endpoint here?

File: backend/main.go [hunk 2]
```diff
@@ -51,12 +51,27 @@ func main() {
 	http.HandleFunc("/api/login", LoginEndpoint)
 	http.HandleFunc("/api/logout", LogoutEndpoint)
 	http.HandleFunc("/api/register", RegisterEndpoint)
+	// TODO handle this in another way
+	// No URL params
 	http.HandleFunc("/api/room", func(w http.ResponseWriter, r *http.Request) {
-		if r.Method == "GET" { // /api/room/:id and /api/room/:id/leave
-			http.Error(w, errorJson("Not Implemented!"), http.StatusNotImplemented) // TODO
-		} else if r.Method == "POST" { // /api/room and /api/room/:id
+		println(r.Method)
```
Comment (retrixe) [inline backend/main.go:57]:
```suggestion
```

File: frontend/imports/components/startWatchingDialog.tsx [hunk 3]
```diff
@@ -45,7 +46,10 @@ const StartWatchingDialog = (props: { shown: boolean, handleClose: () => void })
           query: fileUrl ? { fileUrl } : {}
         }, `/room/${res.id}`).catch(console.error)
       }
-    } catch (e) { setError('An unknown network error occurred.') }
+    } catch (e) {
+      console.log(e)
+      setError('An unknown network error occurred.')
+    }
```
Comment (retrixe) [inline frontend/imports/components/startWatchingDialog.tsx:52]: This doesn't seem like a relevant change?

File: backend/rooms.go [hunk 4]
```diff
@@ -1 +1,77 @@
...
+		return
+	} else if rows, err := result.RowsAffected(); err != nil || rows != 1 {
+		handleInternalServerError(w, err)
+		return
+	}
+	w.Write([]byte("{\"id\":\"" + uuid.String() + "\"}"))
+}
+
+func GetRoom(w http.ResponseWriter, r *http.Request) {
+	// TODO: Allow guests
```
Comment (retrixe) [inline backend/rooms.go:55]: This is more of a feature request imo than a TODO that needs doing at the moment

File: backend/rooms.go [hunk 5]
```diff
@@ -1 +1,77 @@
...
+	err := findRoomByIdStmt.QueryRow(id).Scan(
+		&room.ID, &room.Type, &room.Title, &room.Extra,
+		pq.Array(&room.Chat), pq.Array(&room.Members), &room.Paused, &room.Timestamp,
+		&room.CreatedAt, &room.LastActionTime)
+	if err != nil {
+		handleInternalServerError(w, err)
+		return
+	}
+	json, _ := json.Marshal(room)
+	w.Write(json)
```
Comment (retrixe) [inline backend/rooms.go:76]:
```suggestion
	json.NewEncoder(w).Encode(room)
```

**Review by retrixe — CHANGES_REQUESTED** [review]

File: backend/main.go (hunk 1 above)
Comment (tierrif) [reply backend/main.go:72]: This will be changed to PATCH instead for updating the room

**Pull request merged**